tests/rt.sh

```diff
@@ -30,77 +30,405 @@ usage() {
+        [[ -n $FAIL_LOG ]] && FAILED_COMPILE_LOGS+=("${FAIL_LOG}")
+      fi
+
+    elif [[ $line =~ RUN ]]; then
+      
+      RMACHINES=$(echo "$line" | cut -d'|' -f3 | sed -e 's/^ *//' -e 's/ *$//')
+      TEST_NAME=$(echo "$line" | cut -d'|' -f2 | sed -e 's/^ *//' -e 's/ *$//')
+      GEN_BASELINE=$(echo "$line" | cut -d'|' -f4 | sed -e 's/^ *//' -e 's/ *$//')
+      
+      if [[ ${RMACHINES} == '' ]]; then
+        valid_test=true
+      elif [[ ${RMACHINES} == -* ]]; then
+        [[ ${RMACHINES} =~ ${MACHINE_ID} ]] || valid_test=true
+      elif [[ ${RMACHINES} == +* ]]; then
+        [[ ${RMACHINES} =~ ${MACHINE_ID} ]] && valid_test=true
+      fi
+
+      if [[ $valid_test == true ]]; then
+        TEST_COUNTER=$((TEST_COUNTER+1))
+        GETMEMFROMLOG=""
+        FAIL_LOG=""
+        TEST_RESULT=""
+        TIME_FILE=""
+        TEST_TIME=""
+        RT_TEST_TIME=""
+        RT_TEST_MEM=""
+        if [[ $CREATE_BASELINE == true && $GEN_BASELINE != "baseline" ]]; then
+          TEST_RESULT="SKIPPED (TEST DOES NOT GENERATE BASELINE)"
+        elif [[ ! -f "${LOG_DIR}/run_${TEST_NAME}_${COMPILER}.log" ]]; then
+          TEST_RESULT="MISSING"
+          FAIL_LOG="N/A"
+        elif [[ -f fail_test_${TEST_NAME}_${COMPILER} ]]; then
+          if [[ -f "${LOG_DIR}/rt_${TEST_NAME}_${COMPILER}.log" ]]; then
+            TEST_RESULT="FAIL TO COMPARE"
+            FAIL_LOG="${LOG_DIR}/rt_${TEST_NAME}_${COMPILER}.log"
+          else
+            TEST_RESULT="FAIL TO RUN"
+            FAIL_LOG="${LOG_DIR}/run_${TEST_NAME}_${COMPILER}.log"
+          fi
+        else
+          if grep -q "quota" "${LOG_DIR}/run_${TEST_NAME}_${COMPILER}.log"; then
+            TEST_RESULT="FAIL FROM DISK QUOTA"
+            FAIL_LOG="${LOG_DIR}/run_${TEST_NAME}_${COMPILER}.log"
+          elif grep -q "timeout" "${LOG_DIR}/run_${TEST_NAME}_${COMPILER}.log"; then
+            TEST_RESULT="FAIL FROM TIMEOUT"
+            FAIL_LOG="${LOG_DIR}/run_${TEST_NAME}_${COMPILER}.log"
+          else
+            
+            TEST_RESULT="PASS"
+            TIME_FILE="${LOG_DIR}/run_${TEST_NAME}_${COMPILER}_timestamp.txt"
+            GETMEMFROMLOG=$(grep "The maximum resident set size" "${LOG_DIR}/rt_${TEST_NAME}_${COMPILER}.log")
+            RT_TEST_MEM=$(echo "${GETMEMFROMLOG:9:${#GETMEMFROMLOG}-1}" | tr -dc '0-9')
+            RT_TEST_MEM=$((RT_TEST_MEM/1000))
+            if [[ -f "${TIME_FILE}" ]]; then
+              while read -r times || [ "$times" ]; do
+                  times="${times#"${times%%[![:space:]]*}"}"
+
+                  DATE1=$(echo "$times" | cut -d ',' -f2 )
+                  DATE2=$(echo "$times" | cut -d ',' -f3 )
+                  DATE3=$(echo "$times" | cut -d ',' -f4 )
+                  DATE4=$(echo "$times" | cut -d ',' -f5 )
+
+                  TEST_TIME=$(date --date=@$((DATE3 - DATE2)) +'%M:%S')
+                  RT_TEST_TIME=$(date --date=@$((DATE4 - DATE1)) +'%M:%S')
+
+              done < "$TIME_FILE"
+            fi
```

Comment:
@DusanJovic-NOAA same here, I'm not sure I understand what time is for what in there. Could you double check I have it right, or what corrections I might need?

Reply:
Looks good. First and last timestamp are starting and ending time of the run_test script, while second and third are starting and ending time of the corresponding test job card script. Same for the run_compile script and the compile job.